sky/clouds/lambda_labs.py

```diff
@@ -0,0 +1,193 @@
+"""Lambda."""
+import json
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from sky import clouds
+from sky.clouds import service_catalog
+from sky.skylet.providers.lambda_labs.lambda_utils import LambdaLabsClient, LambdaLabsError
+
+if typing.TYPE_CHECKING:
+    # Renaming to avoid shadowing variables.
+    from sky import resources as resources_lib
+
+# Minimum set of files under ~/.lambda_labs that grant Lambda Labs access.
+_CREDENTIAL_FILES = [
+    'lambda_keys',
+]
+
+
+@clouds.CLOUD_REGISTRY.register
+class Lambda(clouds.Cloud):
+    """Lambda Labs GPU Cloud."""
+
+    @classmethod
+    def regions(cls):
+        cls._regions = [
+            clouds.Region('us-east-1'),
+            clouds.Region('us-west-2'),
+            clouds.Region('australia-southeast-1'),
+            clouds.Region('europe-central-1'),
+            clouds.Region('asia-south-1'),
+            clouds.Region('me-west-1'),
+            clouds.Region('europe-south-1'),
+            clouds.Region('asia-northeast-1'),
+            clouds.Region('asia-northeast-2'),
+            clouds.Region('us-west-1'),
+            clouds.Region('us-south-1'),
+        ]
+        return cls._regions
+
+    @classmethod
+    def region_zones_provision_loop(
+        cls,
+        *,
+        instance_type: Optional[str] = None,
+        accelerators: Optional[Dict[str, int]] = None,
+        use_spot: bool,
+    ) -> Iterator[Tuple[clouds.Region, List[clouds.Zone]]]:
+        del instance_type
+        del use_spot
+        del accelerators  # unused
+
+        for region in cls.regions():
+            yield region, region.zones
```

Comment:
Hmmm, ok I will merge master first

Reply:
`service_catalog.get_region_zones_for_instance_type(instance_type, use_spot, 'lambda')`  returns regions in alphabetical order, which is annoying since `us-east-1` and `us-west-2` have the most availability.

Reply:
For that problem, do you think we can do a similar hack as https://github.com/skypilot-org/skypilot/blob/8d6f6a9ce1d7530c7adc48e80b4154b63220c180/sky/clouds/service_catalog/aws_catalog.py#L125-L133

Reply:
I think that is possible. However, to the best of my knowledge, all instances are offered in all regions, so returning `cls.regions()` works just fine.

Reply:
I am trying to make it easier for future refactoring, i.e. decoupling the offering information with the cloud implementation. It might be better to leave the region_zones information in the catalog and `cls.regions()` should be deprecated. Wdyt @ewzeng?

Reply:
Ah I see. Sure, I can try to not use `cls.regions()`.

Reply:
Fixed!